### PR TITLE
Ballchasy - bcManager update

### PR DESCRIPTION
**Pull request by BenjaminHunt**
# bcManager
- re-writes all `bcManager` code
- uses python-ballchasing api wrapper (https://github.com/Rolv-Arild/python-ballchasing)
- uses roles for ballchasing tier name hierarchy (1Premier, 2Master, etc.)
- includes customizable timezones (used for ballchasing searches, Default: `"America/New_York"`, Reference: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)
- posts and updates a status embed for `reportMatches` command while processing match replays
- includes game and match validations
- posts score report to score reporting channel
- updates match cog info after score is reported
- `<p>bcr` processes a player's match results (temporary)
- `<p>reportMatches` processes all match replays for current match day

# match
- updates add match commands to require `match_type` (Regular season, wildcard, etc) and `match_format` (4-gs, bo-5, etc)
- removes lobby info inputs in add match behavior
- adds lobby info validation to avoid collisions on match days

### CODE REVIEW

File: bcManager/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-from .config import config
+from .bcConfig import bcConfig
```
Comment (blister) [inline bcManager/__init__.py:2]: Tiny nitpick, but the naming convention of the module has capital `BC` directly below this, and you went with lower case `bc`. Feels incorrect. Obviously it doesn't matter in code, but I love consistency.

File: bcManager/bcConfig.py [hunk 2]
```diff
@@ -3,23 +3,18 @@
...
-    top_level_group = None
-    search_count = 10
+class bcConfig:
+    # Search Settings
+    SEARCH_COUNT = 10
+    PLAYLIST = "private"
+    SORT_BY = "replay-date"
+    SORT_DIR = "desc"
+
+    ZONE_ADJ = "-04:00"
```
Comment (blister) [inline bcManager/bcConfig.py:13]: This zone adjustment doesn't take into account daylight savings time. Part of the year has Eastern at EST `-5`.

**Review by blister — CHANGES_REQUESTED** [review]
Overall, code looks great. One minor nitpick in naming convention and one bug that will pop up on November 6th when the time changes.

**Review by blister — APPROVED** [review]
Nicely done. Timezones are always a pain in the ass. :(

**Review by blister — APPROVED** [review]

**Review by blister — APPROVED** [review]
I concur!

**Review by blister — APPROVED** [review]
Trying again.

**Comment by BenjaminHunt** [comment]
`BCConfig.py` name wasn't updated, so that was added as a direct commit to master: https://github.com/RSC-NA/bot-RSCBot/commit/cd027b3ee1151db745e1855863a80f44fd16d3fa

**Comment by BenjaminHunt** [comment]
another sloppy miss, but we're good now. Backwards compatibility patch: https://github.com/RSC-NA/bot-RSCBot/commit/7f7be591da0198b0bb24843f92a3ef7a446599a4

**Comment by BenjaminHunt** [comment]
well at least match is cleaned up... https://github.com/RSC-NA/bot-RSCBot/commit/6ff084e1e4d653969339bd2b2d9530695a294d80

**Comment by BenjaminHunt** [comment]
+ https://github.com/RSC-NA/bot-RSCBot/commit/6ff084e1e4d653969339bd2b2d9530695a294d80

**Comment by BenjaminHunt** [comment]
+ time format bad https://github.com/RSC-NA/bot-RSCBot/commit/b9361bc44a78b209bfbf5b0c14d217c219dc767e

**Comment by BenjaminHunt** [comment]
With all my heart, I believe this to be the last panic fix: https://github.com/RSC-NA/bot-RSCBot/commit/1ad7df4ceb90ae0516ca9a334f195da19699dec0

**Pull request merged**